server.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+
+const express = require("express");
+const favicon = require("serve-favicon");
+const bodyParser = require("body-parser");
+const session = require("express-session");
+// const csrf = require('csurf');
+const consolidate = require("consolidate"); // Templating library adapter for Express
+const swig = require("swig");
+// const helmet = require("helmet");
+const MongoClient = require("mongodb").MongoClient; // Driver for connecting to MongoDB
+const http = require("http");
+const marked = require("marked");
+//const nosniff = require('dont-sniff-mimetype');
+const app = express(); // Web framework to handle routing requests
+const routes = require("./app/routes");
+const { port, db, cookieSecret } = require("./config/config"); // Application config properties
+/*
+// Fix for A6-Sensitive Data Exposure
+// Load keys for establishing secure HTTPS connection
+const fs = require("fs");
+const https = require("https");
+const path = require("path");
+const httpsOptions = {
+    key: fs.readFileSync(path.resolve(__dirname, "./artifacts/cert/server.key")),
+    cert: fs.readFileSync(path.resolve(__dirname, "./artifacts/cert/server.crt"))
+};
+*/
+
+MongoClient.connect(db, (err, db) => {
+    if (err) {
+        console.log("Error: DB: connect");
+        console.log(err);
+        process.exit(1);
+    }
+    console.log(`Connected to the database`);
+
+    /*
+    // Fix for A5 - Security MisConfig
+    // TODO: Review the rest of helmet options, like "xssFilter"
+    // Remove default x-powered-by response header
+    app.disable("x-powered-by");
+
+    // Prevent opening page in frame or iframe to protect from clickjacking
+    app.use(helmet.frameguard()); //xframe deprecated
+
+    // Prevents browser from caching and storing page
+    app.use(helmet.noCache());
+
+    // Allow loading resources only from white-listed domains
+    app.use(helmet.contentSecurityPolicy()); //csp deprecated
+
+    // Allow communication only on HTTPS
+    app.use(helmet.hsts());
+
+    // TODO: Add another vuln: https://github.com/helmetjs/helmet/issues/26
+    // Enable XSS filter in IE (On by default)
+    // app.use(helmet.iexss());
+    // Now it should be used in hit way, but the README alerts that could be
+    // dangerous, like specified in the issue.
+    // app.use(helmet.xssFilter({ setOnOldIE: true }));
+
+    // Forces browser to only use the Content-Type set in the response header instead of sniffing or guessing it
+    app.use(nosniff());
+    */
+
+    // Adding/ remove HTTP Headers for security
+    app.use(favicon(__dirname + "/app/assets/favicon.ico"));
+
+    // Express middleware to populate "req.body" so we can access POST variables
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({
+        // Mandatory in Express v4
+        extended: false
+    }));
+
+    // Enable session management using express middleware
+    app.use(session({
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- f14a617b6287b4de3739c9bdd5120b12 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Sensitive Cookie with Improper SameSite Attribute
_Low CWE-1275_
The `cookie.sameSite` option is either undefined or set to the insecure value `none` or `false`. In all three cases, the cookie's `SameSite` attribute protections are disabled, allowing the browser to send cookies in cross-site requests. An attacker can abuse this missing protection to facilitate cross-site request forgery (CSRF) attacks.



### How to fix
The server should explicitly set the SameSite attribute for all cookies containing sensitive data, such as session tokens. This can be accomplished either programmatically when setting Set-Cookie headers in responses or via deployment descriptors provided by various frameworks.

The setting "SameSite=Strict" provides the strongest protection against CSRF but the user experience works best when an application uses two different cookies, one for write-access that is locked down with "SameSite=Strict" and another for read-access that can have a more permissive setting, e.g., "SameSite=Lax". If the application only relies on one cookie for both read & write access, then the setting "SameSite=Lax" provides some protection against CSRF without too much negative impact on user experience. 

It's worth noting that the definition of same-site is less restrictive than the definition of cross-origin, e.g., requests from other sub-domains are considered same-site although they are from a different origin. This means that even in browsers that fully support the "SameSite" cookie attribute, an attacker may still circumvent this protection if they can take over a subdomain or inject HTML or JavaScript into any subdomain of the same site. Therefore, relying on the "SameSite" attribute is not recommended as the sole defense against cross-site request forgery, it should instead be seen as a defense in depth mechanism that reduces the likelihood of CSRF.